minio-admin/deployment/src/main/java/io/quarkiverse/minio/admin/client/deployment/MinioAdminClientNativeProcessor.java

```diff
@@ -0,0 +1,66 @@
+package io.quarkiverse.minio.admin.client.deployment;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.IndexDependencyBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+
+public class MinioAdminClientNativeProcessor {
+
+    @BuildStep
+    void addDependencies(BuildProducer<IndexDependencyBuildItem> indexDependency) {
+        indexDependency.produce(new IndexDependencyBuildItem("io.minio", "minio-admin"));
+    }
+
+    @BuildStep
+    void registerForReflection(
+            BuildProducer<ReflectiveClassBuildItem> reflectionClasses) {
+        List<String> classes = getClasses("io.minio.admin.messages", null);
+        classes.addAll(getClasses("io.minio.admin.messages.info", null));
+        classes.addAll(getClasses("io.minio.admin", ".*Resp$|.*Info$"));
+        reflectionClasses
+                .produce(ReflectiveClassBuildItem.builder(classes.toArray(new String[0])).fields(true).methods(true).build());
+    }
+
+    private List<String> getClasses(String packageName, String regexp) {
+        try {
+            ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+            assert classLoader != null;
+            String path = packageName.replace('.', '/');
+            URI uri = Objects.requireNonNull(classLoader.getResource(path)).toURI();
+            try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object> emptyMap())) {
+                Path packagePath = fileSystem.getPath(path);
+                List<Path> dirs = Files.walk(packagePath, 1).collect(Collectors.toList());
+                return dirs.stream().map(item -> findClasses(item, packageName, regexp))
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toList());
+            }
+        } catch (IOException | URISyntaxException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private String findClasses(Path path, String packageName, String regexp) {
+        String fileName = path.getFileName().toString();
+        if (fileName.endsWith(".class")) {
+            String className = fileName.substring(0, fileName.length() - 6);
+            if (regexp == null || className.matches(regexp)) {
+                return packageName + '.' + className;
+            }
+        }
+        return null;
+    }
```

Comment:
Ok, so it seems my previous work on native as really, really lame....

```suggestion
    private List<String> getClasses(CombinedIndexBuildItem index, String packageName) {
        return getClasses(index, packageName, ".*");
    }

    private List<String> getClasses(CombinedIndexBuildItem index, String packageName, String regexp) {
        return index.getIndex().getClassesInPackage(packageName).stream()
                .filter(c -> c.name().toString().matches(regexp))
                .map(c -> c.name().toString())
                .toList();
    }
```

Invocation needs to be adapted accordingly.